### ISSUE TITLE
ob run not working for macOS 12.0.1

### ISSUE THREAD

**Issue by dbarkleyjr** [obsidiansystems/obelisk#895]
I followed the [installation instructions](https://github.com/obsidiansystems/obelisk#installing-obelisk) exactly and when I go to run `ob run` after `ob init` I get:

```
ob: dieVerbatim: user error (ob: Cannot find the program 'ghc'. User-specified path
'/Users/dbarkleyjr/Projects/reflex' does not refer to an
executable and the program is not on the system path.
)
```

I'm on macOS 12.0.1. I didn't see anything about explicitly installing ghc.

**Comment by dustinnorwood** [comment]
Hi @dbarkleyjr, I am having the same issue. Were you able to find a solution?

**Comment by dbarkleyjr** [comment]
> Hi @dbarkleyjr, I am having the same issue. Were you able to find a solution?

Nope. I decided to just use purescript for my frontend. Much less friction setting it up

**Comment by romefeller** [comment]
This error happens when you use zsh instead of bash. The nixpkgs was not set (nix-channel --add https://nixos.org/channels/nixpkgs-21.11-darwin nixpkgs" and you will need to set it up the env var yourself. You can also try "nix-shell -p 'bash'" and run again "ob run". We will be working on this carefully.

Thank you!

**Comment by madeline-os** [comment]
We need to either document the need for a channel and env var set up, or make the installation easier.

**Comment by coelhomarcio** [comment]
I'm using _bash_ and facing the same issue...

```
$ nix-env --version
nix-env (Nix) 2.8.1
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 9.0.2
$ ob run
./.obelisk/impl: command not cached, building ...
Tip: Press Ctrl+e to display full output
✔ Built on ./.obelisk/impl [command]
ob: dieVerbatim: user error (ob: Cannot find the program 'ghc'. User-specified path
'/path/for/my/project' does not refer to an
executable and the program is not on the system path.
)
$ 
```

**Comment by romefeller** [comment]
Is your NIX_PATH empty? "printenv NIX_PATH" to check.

**Comment by coelhomarcio** [comment]
> Is your NIX_PATH empty? "printenv NIX_PATH" to check.

I read somewhere that on non NixOs, the NIX_PATH doesn't get set anymore. But it worked after I add/update nix-channel and export the NIX_PATH env var. And yes, my NIX_PATH was empty.
Thanks for your help!

**Comment by madeline-os** [comment]
The solution to this issue will need to remove all dependencies on NIX_PATH that obelisk may have in non-nixos environments.
